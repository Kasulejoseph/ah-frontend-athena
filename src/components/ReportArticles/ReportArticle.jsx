import React from "react";
import {
  MDBContainer,
  MDBModal,
  MDBIcon,
  MDBModalBody,
  MDBModalHeader
} from "mdbreact";
import PropTypes from "prop-types";
import ReportView from "../../views/ReportArticles/ReportView";

export const ReportArticle = ({ modal, close, article }) => {
  return (
    <MDBContainer>
      <MDBModal isOpen={modal} className="">
        <MDBIcon
          icon="times"
          onClick={close}
          style={{ right: 5, position: "absolute" }}
          className=""
        />
        <MDBModalHeader>
          Please tell us why you are reporting this article
        </MDBModalHeader>

        <MDBModalBody>
          <ReportView article={article} />
        </MDBModalBody>
      </MDBModal>
    </MDBContainer>
  );
};

ReportArticle.propTypes = {
  modal: PropTypes.bool.isRequired,
  close: PropTypes.func.isRequired,
  article: PropTypes.shape({})
};

ReportArticle.defaultProps = {
  article: {}
};

export default ReportArticle;
