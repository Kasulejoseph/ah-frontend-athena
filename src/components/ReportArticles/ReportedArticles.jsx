import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { GetReportedAction } from "../../actions/ReportArticleAction";
import "./ReportedArticles.scss";

export class ReportedArticles extends Component {
  constructor() {
    super();
    this.state = {
      reportedArticles: []
    };
  }

  componentDidMount() {
    const { GetReportedAction } = this.props;
    GetReportedAction();
  }

  componentWillReceiveProps(nextProps) {
    const { reportedArticles } = nextProps;
    if (reportedArticles) {
      this.setState({
        reportedArticles: reportedArticles
      });
    }
  }

  render() {
    const capitalizeFirstLetter = string => {
      return string.charAt(0).toUpperCase() + string.slice(1);
    };
    const returnArticleTitle = slug => {
      const titleArray = slug.split("-");
      titleArray.pop();
      return capitalizeFirstLetter(titleArray.join(" "));
    };

    const { reportedArticles } = this.state;
    const list = reportedArticles.map(item => {
      let title = returnArticleTitle(item.article_slug);
      return (
        <li key={item.article_id} className="mt-5 container grey-bg">
          <div className="p-1">
            <th>
              <h2>{title}</h2>
            </th>
            <table>
              <tr>
                <td>
                  <strong>Reason:</strong>
                </td>
                <td>
                  <strong>Date:</strong>
                </td>
                <td>
                  <strong>Username:</strong>
                </td>
                <td>
                  <strong>Email:</strong>
                </td>
              </tr>

              <tr>
                <td>{item.reason}</td>
                <td>{item.reported_at}</td>
                <td>{item.reported_by.username}</td>
                <td>{item.reported_by.email}</td>
              </tr>
            </table>
          </div>
        </li>
      );
    });
    if (list.length === 0) {
      return (
        <div className="text-center">
          <h2>
            Either there are no reported articles or you are not authorized to
            view this page
          </h2>
        </div>
      );
    } else
      return (
        <ul className="mt-5 d-flex flex-column">
          <h1 className="text-center">Reported Articles</h1>
          {list}
        </ul>
      );
  }
}

export const mapStateToProps = state => {
  return {
    reportedArticles: state.report.reportedArticles.articles
  };
};

ReportedArticles.propTypes = {
  GetReportedAction: PropTypes.func.isRequired,
  reportedArticles: PropTypes.shape({}).isRequired
};

export default connect(
  mapStateToProps,
  {
    GetReportedAction
  }
)(ReportedArticles);
