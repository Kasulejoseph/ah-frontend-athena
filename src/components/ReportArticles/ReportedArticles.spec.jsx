import React from "react";
import { shallow } from "enzyme";
import { ReportArticle } from "./ReportArticle";
import { ReportedArticles, mapStateToProps } from "./ReportedArticles";

let props = { modal: {}, close: {}, article: {} };
let props2 = {
  GetReportedAction: jest.fn(),
  reportedArticles: {}
};
let wrapper;
let wrap;

describe("test report article component", () => {
  beforeEach(() => {
    wrapper = shallow(<ReportArticle {...props} />);
  });
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("test reported articles", () => {
  beforeEach(() => {
    wrap = shallow(<ReportedArticles {...props2} />);
  });
  it("should match snapshot", () => {
    expect(wrap).toMatchSnapshot();
  });
  it("will receive props", () => {
    wrap.setProps({
      reportedArticles: [
        {
          article_id: 1,
          article_slug: "i-was-the-first-05a811f9f44e",
          reported_by: {
            username: "lugada",
            bio: "",
            image: "",
            email: "lugjosh@gmail.com"
          },
          reason: "this article does not look good",
          reported_at: "2019-02-01T08:53:29.993268Z"
        }
      ]
    });
    expect(wrap.instance().state).toEqual({
      reportedArticles: [
        {
          article_id: 1,
          article_slug: "i-was-the-first-05a811f9f44e",
          reason: "this article does not look good",
          reported_at: "2019-02-01T08:53:29.993268Z",
          reported_by: {
            bio: "",
            email: "lugjosh@gmail.com",
            image: "",
            username: "lugada"
          }
        }
      ]
    });
  });
  it("should map state to props", () => {
    mapStateToProps({report: {reportedArticles: {reportedArticles:{}}}})
  });
});
