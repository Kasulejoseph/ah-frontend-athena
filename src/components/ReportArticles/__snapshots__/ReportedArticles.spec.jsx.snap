// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test report article component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReportArticle
    article={Object {}}
    close={Object {}}
    modal={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <t
        autoFocus={true}
        backdrop={true}
        backdropTransitionTimeout={150}
        className=""
        fade={true}
        isOpen={Object {}}
        modalTransitionTimeout={300}
        role="dialog"
        tabIndex="-1"
        zIndex={1050}
      >
        <t
          border={false}
          brand={false}
          className=""
          fab={false}
          fal={false}
          far={false}
          fixed={false}
          flip=""
          icon="times"
          inverse=""
          light={false}
          list={false}
          onClick={Object {}}
          pull=""
          pulse={false}
          regular={false}
          rotate=""
          size=""
          spin={false}
          stack=""
          style={
            Object {
              "position": "absolute",
              "right": 5,
            }
          }
        />
        <t
          closeAriaLabel="Close"
          tag="h4"
        >
          Please tell us why you are reporting this article
        </t>
        <t>
          <Connect(ReportView)
            article={Object {}}
          />
        </t>
      </t>,
      "fluid": false,
      "tag": "div",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoFocus": true,
        "backdrop": true,
        "backdropTransitionTimeout": 150,
        "children": Array [
          <t
            border={false}
            brand={false}
            className=""
            fab={false}
            fal={false}
            far={false}
            fixed={false}
            flip=""
            icon="times"
            inverse=""
            light={false}
            list={false}
            onClick={Object {}}
            pull=""
            pulse={false}
            regular={false}
            rotate=""
            size=""
            spin={false}
            stack=""
            style={
              Object {
                "position": "absolute",
                "right": 5,
              }
            }
          />,
          <t
            closeAriaLabel="Close"
            tag="h4"
          >
            Please tell us why you are reporting this article
          </t>,
          <t>
            <Connect(ReportView)
              article={Object {}}
            />
          </t>,
        ],
        "className": "",
        "fade": true,
        "isOpen": Object {},
        "modalTransitionTimeout": 300,
        "role": "dialog",
        "tabIndex": "-1",
        "zIndex": 1050,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "border": false,
            "brand": false,
            "className": "",
            "fab": false,
            "fal": false,
            "far": false,
            "fixed": false,
            "flip": "",
            "icon": "times",
            "inverse": "",
            "light": false,
            "list": false,
            "onClick": Object {},
            "pull": "",
            "pulse": false,
            "regular": false,
            "rotate": "",
            "size": "",
            "spin": false,
            "stack": "",
            "style": Object {
              "position": "absolute",
              "right": 5,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Please tell us why you are reporting this article",
            "closeAriaLabel": "Close",
            "tag": "h4",
          },
          "ref": null,
          "rendered": "Please tell us why you are reporting this article",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(ReportView)
              article={Object {}}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "article": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <t
          autoFocus={true}
          backdrop={true}
          backdropTransitionTimeout={150}
          className=""
          fade={true}
          isOpen={Object {}}
          modalTransitionTimeout={300}
          role="dialog"
          tabIndex="-1"
          zIndex={1050}
        >
          <t
            border={false}
            brand={false}
            className=""
            fab={false}
            fal={false}
            far={false}
            fixed={false}
            flip=""
            icon="times"
            inverse=""
            light={false}
            list={false}
            onClick={Object {}}
            pull=""
            pulse={false}
            regular={false}
            rotate=""
            size=""
            spin={false}
            stack=""
            style={
              Object {
                "position": "absolute",
                "right": 5,
              }
            }
          />
          <t
            closeAriaLabel="Close"
            tag="h4"
          >
            Please tell us why you are reporting this article
          </t>
          <t>
            <Connect(ReportView)
              article={Object {}}
            />
          </t>
        </t>,
        "fluid": false,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": true,
          "backdrop": true,
          "backdropTransitionTimeout": 150,
          "children": Array [
            <t
              border={false}
              brand={false}
              className=""
              fab={false}
              fal={false}
              far={false}
              fixed={false}
              flip=""
              icon="times"
              inverse=""
              light={false}
              list={false}
              onClick={Object {}}
              pull=""
              pulse={false}
              regular={false}
              rotate=""
              size=""
              spin={false}
              stack=""
              style={
                Object {
                  "position": "absolute",
                  "right": 5,
                }
              }
            />,
            <t
              closeAriaLabel="Close"
              tag="h4"
            >
              Please tell us why you are reporting this article
            </t>,
            <t>
              <Connect(ReportView)
                article={Object {}}
              />
            </t>,
          ],
          "className": "",
          "fade": true,
          "isOpen": Object {},
          "modalTransitionTimeout": 300,
          "role": "dialog",
          "tabIndex": "-1",
          "zIndex": 1050,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "border": false,
              "brand": false,
              "className": "",
              "fab": false,
              "fal": false,
              "far": false,
              "fixed": false,
              "flip": "",
              "icon": "times",
              "inverse": "",
              "light": false,
              "list": false,
              "onClick": Object {},
              "pull": "",
              "pulse": false,
              "regular": false,
              "rotate": "",
              "size": "",
              "spin": false,
              "stack": "",
              "style": Object {
                "position": "absolute",
                "right": 5,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Please tell us why you are reporting this article",
              "closeAriaLabel": "Close",
              "tag": "h4",
            },
            "ref": null,
            "rendered": "Please tell us why you are reporting this article",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Connect(ReportView)
                article={Object {}}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "article": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`test reported articles should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReportedArticles
    GetReportedAction={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    reportedArticles={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h2>
        Either there are no reported articles or you are not authorized to view this page
      </h2>,
      "className": "text-center",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Either there are no reported articles or you are not authorized to view this page",
      },
      "ref": null,
      "rendered": "Either there are no reported articles or you are not authorized to view this page",
      "type": "h2",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
          Either there are no reported articles or you are not authorized to view this page
        </h2>,
        "className": "text-center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Either there are no reported articles or you are not authorized to view this page",
        },
        "ref": null,
        "rendered": "Either there are no reported articles or you are not authorized to view this page",
        "type": "h2",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
